src/ui/colorui.js

```diff
@@ -117,7 +135,60 @@ export default class ColorUI extends Plugin {
 				editor.editing.view.focus();
 			} );
 
+			if ( documentColorsCount !== 0 ) {
+				dropdownView.on( 'change:isOpen', ( evt, name, val ) => {
+					if ( val ) {
+						const model = editor.model;
+						const document = model.document;
+						const rootNames = document.getRootNames();
+						const maxCount = this.colorTableView.documentColorsCount;
+						const documentColors = this.colorTableView.documentColors;
+						documentColors.clear();
+
+						for ( const rootName of rootNames ) {
+							const root = document.getRoot( rootName );
+							const range = model.createRangeIn( root );
+							for ( const node of range.getItems() ) {
+								if ( node.is( 'textProxy' ) && node.hasAttribute( this.componentName ) ) {
+									this.addColorToDocumentColors( node.getAttribute( this.componentName ) );
+									if ( documentColors.length >= maxCount ) {
+										break;
+									}
+								}
+							}
+							if ( documentColors.length >= maxCount ) {
```

Comment:
I don't see any reason for this check here. New colors are only added in line https://github.com/ckeditor/ckeditor5-font/blob/47da79a/src/ui/colorui.js#L159 and current `documentColors` size is correctly checked afterwards.

This check is redundant.

Reply:
This check was made to exit from outer `for` loop, to not iterate over other roots in editor, when already all document colors are found.
I'll extract this part to function and made single `return` statement to increase its readability and remove additional `if` statement.

Reply:
Sounds good 👍